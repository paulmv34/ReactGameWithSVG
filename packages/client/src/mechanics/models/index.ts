export { Entity, EntityEvent } from './Entity/Entity'
export { EntityDynamic } from './EntityDynamic/EntityDynamic'
export { Flag } from './Flag/Flag'
export { Bullet } from './Bullet/Bullet'
export { Tank } from './Tank/Tank'
export { TankEnemy } from './Tank/TankEnemy'
export {
  type PlayerVariant,
  type TankPlayerSettings,
  TankPlayer,
} from './Tank/TankPlayer'
export { Terrain } from './Terrain/Terrain'
export { Powerup } from './Powerup/Powerup'
export { Score } from './Score/Score'
export { Explosion } from './Explosion/Explosion'
