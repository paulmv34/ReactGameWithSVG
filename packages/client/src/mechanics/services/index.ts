export { ControllerKeyboard } from './Controller/ControllerKeyboard'
export { ControllerManager } from './Controller/ControllerManager'
export { type GameMode, Game } from './Game/Game'
export { Loop } from './Loop/Loop'
export { MapManager } from './MapManager/MapManager'
export { Resources, ResourcesEvent } from './Resources/Resources'
export { Scenario, ScenarioEvent } from './Scenario/Scenario'
export { State } from './State/State'
export { View } from './View/View'
export { Zone } from './Zone/Zone'
