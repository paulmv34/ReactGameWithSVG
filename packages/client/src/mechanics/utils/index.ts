export { EventEmitter } from './EventEmitter/EventEmitter'
export { isKeyOfObject } from './isKeyOfObject'
export { rand } from './rand'
export { sleep } from './sleepTimer'
