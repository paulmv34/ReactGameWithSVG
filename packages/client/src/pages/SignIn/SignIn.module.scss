.title {
  margin-bottom: 64px;

  font-size: 2rem;
  color: var(--color-text-dark);
}
