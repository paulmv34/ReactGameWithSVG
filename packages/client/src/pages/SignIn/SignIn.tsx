import { FC, useEffect } from 'react'
import styles from './SignIn.module.scss'
import clsx from 'clsx'
import { useLocation, useNavigate } from 'react-router'

import { ROUTES } from '@/types/types'

import SignInForm from '@/modules/SignInForm/SignInForm'
import store from '@/store/store'

const SignIn: FC = () => {
  const navigate = useNavigate()
  const { state } = useLocation()
  const { isLoggedIn } = store.getState()

  const onAuth = () => {
    navigate(state?.path || ROUTES.MAIN)
  }

  useEffect(() => {
    if (isLoggedIn) {
      navigate(state?.path || ROUTES.MAIN)
    }
  }, [])

  return (
    <section className={clsx(styles.page, 'page')}>
      <div className="wrapper wrapper-backlink">
        <h2 className={clsx(styles.title, 'page-title')}>Вход</h2>
        <SignInForm onAuth={onAuth} />
      </div>
    </section>
  )
}

export default SignIn
