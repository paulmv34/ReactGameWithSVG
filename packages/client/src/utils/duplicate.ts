import { TReaction } from '@/pages/Forum/components/ForumContent/ForumItem/types'

interface IObj {
  [key: string]: number
}
export const duplicate = (data: TReaction[]) => {
  if (!data) {
    return []
  }
  console.log(data)
  const counter: IObj = data?.reduce((obj, i) => {
    const res: IObj = { ...obj }
    if (!Object.prototype.hasOwnProperty.call(res, i.emoji)) {
      res[i.emoji] = 0
    }
    res[i.emoji]++
    return res
  }, {})

  const currentArr = Object.keys(counter)?.map((emoji) => {
    return { emoji: emoji, count: counter[emoji] }
  })

  return currentArr
}
