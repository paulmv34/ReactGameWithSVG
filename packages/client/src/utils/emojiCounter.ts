import { TReaction } from '@/pages/Forum/components/ForumContent/ForumItem/types'

export const emojiCounter = (data: TReaction[]) => {
  if (!data) {
    return []
  }
  const counter: Record<string, number> = data?.reduce((obj, i) => {
    const res: Record<string, number> = { ...obj }
    if (!Object.prototype.hasOwnProperty.call(res, i.emoji)) {
      res[i.emoji] = 0
    }
    res[i.emoji]++
    return res
  }, {})

  const currentArr = Object.keys(counter)?.map((emoji) => {
    return { emoji: emoji, count: counter[emoji] }
  })

  return currentArr
}
