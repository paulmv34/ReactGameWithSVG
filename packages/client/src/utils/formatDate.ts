export const formatDate = (dateString: string) => {
  const date = new Date(dateString)

  return new Intl.DateTimeFormat('ru-RU', { year: 'numeric', month: '2-digit', day: '2-digit' }).format(date)
}
