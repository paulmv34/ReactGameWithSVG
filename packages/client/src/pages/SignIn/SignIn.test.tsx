import React from 'react'
import * as renderer from 'react-test-renderer'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import SignIn from '@/pages/SignIn/SignIn'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { fieldErrorMessages } from '@/utils/validationSchema'
import { sleep } from '@/mechanics/utils'

jest.mock('clsx', () => ({
  default: jest.fn(),
}))

const testApp = (
  <BrowserRouter>
    <Routes>
      <Route path={'*'} element={<SignIn />} />
    </Routes>
  </BrowserRouter>
)

const setup = () => {
  const user = userEvent.setup()
  const formPage = render(testApp)

  const loginInput = screen.getByLabelText('Логин')
  const passwordInput = screen.getByLabelText('Пароль')
  const submitButton = screen.getByRole('button', { name: /войти/i })

  return {
    user,
    formPage,
    loginInput,
    passwordInput,
    submitButton,
  }
}

test('should correctly displayed', async () => {
  const { loginInput, passwordInput, submitButton } = setup()
  expect(loginInput).toBeDefined()
  expect(passwordInput).toBeDefined()
  expect(submitButton).toBeDefined()
})

test('should correctly validate single input interaction', async () => {
  const { loginInput, passwordInput, user } = setup()
  await user.type(loginInput, '111ThisLoginIsIncorrect')
  await user.click(passwordInput)
  expect(screen.getAllByText(fieldErrorMessages.wrongFormat)).toHaveLength(1)
})

test('should correctly validate empty form submit', async () => {
  const { submitButton, user } = setup()
  await user.click(submitButton)
  await sleep(10)
  expect(screen.getAllByText(fieldErrorMessages.required)).toHaveLength(2)
})

test('should correctly validate incorrect form submit', async () => {
  const { loginInput, passwordInput, submitButton, user } = setup()
  await user.type(loginInput, '111ThisLoginIsIncorrect')
  await user.type(passwordInput, 'thispasswordisincorrect')
  await user.click(submitButton)
  await sleep(10)
  expect(screen.getAllByText(fieldErrorMessages.wrongFormat)).toHaveLength(2)
})

test('should submit correct form', async () => {
  const { loginInput, passwordInput, submitButton, user } = setup()
  await user.type(loginInput, 'login')
  await user.type(passwordInput, 'Password1')
  await user.click(submitButton)
  await sleep(200)
})

test('should be equal to snapshot', async () => {
  const tree = renderer.create(testApp).toJSON()
  expect(tree).toMatchSnapshot()
})
