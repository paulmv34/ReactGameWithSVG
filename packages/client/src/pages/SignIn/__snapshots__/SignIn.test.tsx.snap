// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be equal snapshot 1`] = `
<section>
  <div
    className="wrapper wrapper-backlink"
  >
    <h2>
      Вход
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="fields-group"
      >
        <div>
          <label
            htmlFor="login"
          >
            Логин
          </label>
          <input
            autoComplete="off"
            id="login"
            name="login"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Введите логин"
            type="text"
            value=""
          />
          <span />
        </div>
        <div>
          <label
            htmlFor="password"
          >
            Пароль
          </label>
          <input
            autoComplete="off"
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Введите пароль"
            type="password"
            value=""
          />
          <span />
        </div>
      </div>
      <button
        disabled={false}
        type="submit"
      >
        Войти
      </button>
      <a
        href="/sign-up"
        onClick={[Function]}
      >
        Регистрация
      </a>
      <div>
        <p>
          Можно войти с помощью
        </p>
        <button
          type="button"
        >
          Яндекс ID
        </button>
      </div>
    </form>
  </div>
</section>
`;
